main.tf

```
@@ -8,9 +8,22 @@ locals {
 
   create_log_group_role     = var.cloud_watch_logs_role_arn == null
   cloud_watch_logs_role_arn = local.create_log_group_role ? join("", aws_iam_role.this.*.arn) : var.cloud_watch_logs_role_arn
+
+  kms_key_alias  = "terraform-cloudtrail-kms-key"
+  kms_key_id     = var.create_cloudtrail && var.create_kms_key ? module.kms.keys[local.kms_key_alias].arn : var.kms_key_id
```

I think this could be simpler, without the `var.create_cloudtrail` check...

```suggestion
  kms_key_id     = var.create_kms_key ? module.kms.keys[local.kms_key_alias].arn : var.kms_key_id
```

---

The create_cloudtrail variable was added to prevent a user from saying they don't want to create cloudtrail but still end up creating a kms key

---

That's gated already, by using create_clodtrail in the create_keys condition in the kms module

---

Ah, true. I revamped it to be included in the local.create_kms_key instead